OpenRA.Mods.Common/Traits/ReloadAmmoPool.cs

```diff
@@ -0,0 +1,66 @@
+#region Copyright & License Information
+/*
+ * Copyright 2007-2017 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation. For more information,
+ * see COPYING.
+ */
+#endregion
+
+using System.Collections.Generic;
+using System.Linq;
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.Common.Traits
+{
+	[Desc("Reloads an ammo pool.")]
+	public class ReloadAmmoPoolInfo : ConditionalTraitInfo
+	{
+		[Desc("Reload ammo pool with this name.")]
+		public readonly string AmmoPool = "primary";
+
+		[Desc("Reload time in ticks per Count.")]
+		public readonly int Delay = 50;
+
+		[Desc("How much ammo is reloaded after Delay.")]
+		public readonly int Count = 1;
+
+		[Desc("Whether or not reload timer should be reset when ammo has been fired.")]
+		public readonly bool ResetOnFire = false;
+
+		[Desc("Play this sound each time ammo is reloaded.")]
+		public readonly string Sound = null;
+
+		public override object Create(ActorInitializer init) { return new ReloadAmmoPool(this); }
+
+		public override void RulesetLoaded(Ruleset rules, ActorInfo ai)
+		{
+			if (ai.TraitInfos<AmmoPoolInfo>().Count(ap => ap.Name == AmmoPool) != 1)
+				throw new YamlException("ReloadsAmmoPool.AmmoPool requires exactly one AmmoPool with matching Name!");
+
+			base.RulesetLoaded(rules, ai);
+		}
+	}
+
+	public class ReloadAmmoPool : ConditionalTrait<ReloadAmmoPoolInfo>, ITick, INotifyCreated
```

Comment:
Once the reload countdown is moved, can we make this a `PausableConditionTrait` and allow pausing that?

Reply:
Makes sense, will do that.